test/src/sdk-harness/test_projects/storage_map/try_insert.rs

```diff
@@ -0,0 +1,64 @@
+use super::*;
+
+#[macro_export]
+macro_rules! generate_try_insert_tests {
+    ($input_string:expr, $key:expr, $value1:expr, $value2:expr) => {
+        paste::paste! {
+            #[tokio::test]
+            async fn [<try_insert_ $input_string _exists>]() {
+                let instance = test_storage_map_instance().await;
+
+                instance.methods().[<insert_into_ $input_string _map>]($key, $value1).call().await.unwrap();
+
+                let prev = instance.methods().[<get_from_ $input_string _map>]($key).call().await.unwrap().value;
+
+                assert_eq!(prev, Some($value1));
+
+                let result = instance.methods().[<try_insert_into_ $input_string _map>]($key, $value2).call().await.unwrap().value;
+
+                assert_eq!(result, Some($value1));
+
+                let after = instance.methods().[<get_from_ $input_string _map>]($key).call().await.unwrap().value;
+
+                assert_eq!(after, Some($value1));
+            }
+
+            #[tokio::test]
+            async fn [<try_insert_ $input_string _does_not_exist>]() {
+                let instance = test_storage_map_instance().await;
+
+                let prev = instance.methods().[<get_from_ $input_string _map>]($key).call().await.unwrap().value;
+
+                assert_eq!(prev, None);
+
+                let result = instance.methods().[<try_insert_into_ $input_string _map>]($key, $value2).call().await.unwrap().value;
+
+                assert_eq!(result, None);
+
+                let after = instance.methods().[<get_from_ $input_string _map>]($key).call().await.unwrap().value;
+
+                assert_eq!(after, Some($value2));
+            }
+        }
+    };
+}
+
+generate_try_insert_tests!(u64_to_bool, 1, true, false);
```

Comment:
How about a fuzzing macro?

Provide the types to the macro and it goes through all combinations without you having to paste them 1 line at a time below, and have some combinations missing

Reply:
Wouldnt that extend the amount of time taken for the test to run by alot?

Reply:
Probably but we need to ensure that we have coverage.

If a game crashes... it sucks but you don't lose money. If this causes a $$$ loss of funds then rip